### PR TITLE
Django 1.6.9

### PR DESCRIPTION
**Pull request by jgmize**
This supersedes #2634

### CODE REVIEW

**Comment by jgmize** [comment]
I pushed this to demo4 for testing and successfully ran the mcom-tests suite against it. I *think* we're good to go with these changes, but we may want to do a little more manual testing against demo4 before merging to be sure.

File: .gitmodules [hunk 1]
```diff
@@ -61,3 +61,12 @@
 [submodule "vendor-local/src/django-filter"]
 	path = vendor-local/src/django-filter
 	url = https://github.com/alex/django-filter.git
+[submodule "vendor-local/src/django"]
+	path = vendor-local/src/django
+	url = git://github.com/django/django.git
+[submodule "vendor-local/src/bleach"]
```
Comment (pmclanahan) [inline .gitmodules]: Version in vendor too old?

**Comment by pmclanahan** [comment]
Good to have better bleach and the `bleach_tags` helper, but probably not necessary for this. I'm fine w/ it though if it works and stabilizes the tests.

File: .gitmodules (hunk 1 above)
Comment (jgmize) [inline .gitmodules]: Right, I got an error when trying to use it with the latest bleach.

**Comment by jgmize** [comment]
You're probably right about bleach being overkill for plain text emails, and in hindsight I probably shouldn't have bothered, but now that the work is done I don't think it hurts anything to have that extra bit of protection.

File: bedrock/mozorg/helpers/misc.py [hunk 2]
```diff
@@ -517,3 +518,8 @@ def slugify(text):
     trailing whitespace.
     """
     return django_slugify(text)
+
+
+@jingo.register.filter
+def bleach_tags(text):
+    return bleach.clean(text, tags=[], strip=True).replace('&amp;', '&')
```
Comment (pmclanahan) [inline bedrock/mozorg/helpers/misc.py]: Maybe `bleach_tags` should return a safe string so we don't need all these safe filters? Is bleach the thing converting ampersands?

**Comment by pmclanahan** [comment]
I think it'd be nice to mark the return of the new `bleach_tags` as safe (since it is) and remove all of the `safe` filters where it's used. It's not a requirement though.

r+wc

File: bedrock/mozorg/helpers/misc.py (hunk 2 above)
Comment (jgmize) [inline bedrock/mozorg/helpers/misc.py]: I thought about that, and then assumed that the explicit `|safe` was a good thing, but maybe doing that together with renaming to `bleach_safe`?

Yes, bleach.clean() converts ampersands and I didn't see a way to tell it not to, so I just convert them back.

Comment (pmclanahan) [inline bedrock/mozorg/helpers/misc.py]: Sounds good. Marking as safe is:

```python
from jinja2 import Markup

def bleach_safe(text):
    return Markup(bleach.clean(text, tags=[], strip=True).replace('&amp;', '&'))
```

Comment (pmclanahan) [inline bedrock/mozorg/helpers/misc.py]: I'm happy to do this and resubmit if you're busy.

**Pull request merged**